### PR TITLE
Patch 2

### PR DESCRIPTION
**Pull request by dezraj**
Unexpectedly GitHub took me to another branch when I was editing the files and then I had already committed some changes when I noticed it, so here they are in this second pull request.

The description for this pull request is the very same of the other I've just requested today: I have reviewed again some texts and messages in the menus. I think I have changed some messages which are only related to the debug, so maybe they are not really necessary to be changed and can perhaps break something in the code. Every commit has a summary of what was changed. Please check if everything is fine to be merged. Thank you so much in advance for your time and work.

### CODE REVIEW

**Comment by zelurker** [comment]
the pt_br translation was just updated by his author.
Do you plan to make the changes to the po files or do you want to cancel
this PR ?

Le dim. 25 déc. 2022 à 08:07, Elvis Araujo ***@***.***> a
écrit :

**Comment by dezraj** [comment]
Hello! I'm afraid I haven't understood your request. Is it better to edit the po files instead of merging this PR? Will it break something too?

Thank you again for your attention.

**Comment by zelurker** [comment]
Once again : everywhere where you change some text which is passed either
to _("text") or gettext("text") then you must replace the old text by the
new one in all the .po files (1 replaced string / file normally).
So for the very beginning of your 1st patch it means
replace "No speed limit !" by "No speed limit!" in es.po french.po it.po
pt_br.po (all the .po files in the locale directory).
You can either do this with a text editor like notepad++, or use poedit
which is used usually for translations, I'd say that here since it's just
about replacing some strings, a text editor would be faster.
Add all these changes to your pull request and then it will be accepted, I
did it for your 1st pull requests, but here it's too many changes.
Either this or we just drop this pull request.
pt_br.po was just updated by his author, I tried to contact the guy who did
the italian version, no reply yet, and that's here that I discovered that
the guy who did the spanish translation didn't leave his mail in the header
of the file as he should have, so I lost it. Oh well...

To be even more complete, still for your 1st patch, this will give you this
part of es.po :
#: source/sdl/control.c:411
msgid "No speed limit !"
msgstr "Sin limite de velocidad !"
you have the position where the text comes from as a reference if you have
some doubt, here it's from source/sdl/control.c, line 411.
So it will become
msgid "No speed limit!"
That's all, but there are quite a lot of strings to replace this time.

Le mar. 27 déc. 2022 à 20:51, Elvis Araujo ***@***.***> a
écrit :

**Comment by dezraj** [comment]
Thank you so much for guiding me. I have already reviewed all the texts in Raine.pot file and in the es.po translation. I should be reviewing the next files in the next days. I think there are just one or two messages that I couldn't review because I didn't understand the abbreviated words. They are:

6737 msgid "dsw" (perhaps dipswitch?)
6869 msgid "lsb"
6877 msgid "msb"

I think it would be also good to remove the abbreviations here so the translators can more easily translate these strings. But these are the exceptions so far. You can check my work in my patch-2 branch.

I guess with the patch-2 branch I'm working now we could forget this "patch 2" pull request, except that it has a suggestion of rearranging some options in the Video options. Take a look and see if you agree with it. Not much to change but something that I found could make that menu more organized a little. Generally hardware information are placed in the bottom of the menus, such as the "About" information in Raine.

Thank you again for your time and work.

**Comment by zelurker** [comment]
Le mer. 28 déc. 2022 à 08:54, Elvis Araujo ***@***.***> a
écrit :

> Thank you so much for guiding me. I have already reviewed all the texts in
> Raine.pot file and in the es.po translation. I should be reviewing the next
> files in the next days. I think there are just one or two messages that I
> couldn't review because I didn't understand the abbreviated words. They are:
>
> 6737 msgid "dsw" (perhaps dipswitch?)
>
Yes, but you are not supposed to change all the texts, just the ones your
patch already changed.

> 6869 msgid "lsb"
> 6877 msgid "msb"
>
It's about the audio format of audio samples, lsb is less significant byte
first, msb is the other way around : most significant byte first. The
meaning is not super important, there are just 2 ways to encode this, lsb
or msb, if you get garbage here, then choose the other one.

> I think it would be also good to remove the abbreviations here so the
> translators can more easily translate these strings. But these are the
> exceptions so far. You can check my work in my patch-2 branch.
>
> I guess with the patch-2 branch I'm working now we could forget this
> "patch 2" pull request, except that it has a suggestion of rearranging some
> options in the Video options. Take a look and see if you agree with it. Not
> much to change but something that I found could make that menu more
> organized a little. Generally hardware information are placed in the bottom
> of the menus, such as the "About" information in Raine.
>
> Thank you again for your time and work.
>
Don't update raine.pot, raine.pot is generated automatically. I said, the
.po files, not the .pot file.
For video options I didn't see anything related to a reorganization in the
patch, and I didn't try to find it in your mirror, but don't go overkill,
you are not supposed to change everything at the same time, it's easier to
revert things if something goes wrong if things are separated.

**Comment by zelurker** [comment]
And this applies to this part as well, you can check the commit for the updated po files, the script, and your edited patch in 1 commit.

**Comment by dezraj** [comment]
Thank you so much for your work. I decided to review the raine.pot file because I found strings which I hadn't reviewed yet in the code. I will try to add them to the source code files and then create another PR with the few strings left to update so you can check and see how to merge.

By the way, the change in the Video menu selectable options is in my second commit of Video options in this PR, here:
https://github.com/zelurker/raine/pull/50/commits/2bc21c27101a2f2de6bde5c022424a655b0707ed

You can see that I moved the "Video information..." option to the end of the menu to match what is more usual to this kind of information in the emulators. I think this is currently causing an issue in the translation files, line 5251:

msgid "Renderer options"
#, fuzzy
msgid "None"
msgstr "Opciones de renderización"

I'm sorry for not being clearer before. Thank you again for your great work.

**Comment by zelurker** [comment]
there is no msgid "Renderer options" in any of the .po files currently, you
scared me on this one !
And yeah at least the script is done and should be reusable.

Le mer. 28 déc. 2022 à 19:48, Elvis Araujo ***@***.***> a
écrit :

**Comment by zelurker** [comment]
But you are right in that video renderer translation and video information
translation were inverted.
No idea how it happened yet since it searches for the original string, I'll
have to double check on that.

Le mer. 28 déc. 2022 à 22:58, Emmanuel Anne ***@***.***> a
écrit :

**Comment by zelurker** [comment]
Yeah it's normal, the program doesn't know any magic, it's supposed to be
used for cases where a text is just modified, not moved around.
When it moves around in the patch, there is just no way to guess where the
original line was.
For Video information... it looks like this :
-{ _("Video info..."), &do_video, },
-{ _("Renderer options"), &renderer_options },
+{ _("Renderer options..."), &renderer_options },
 { _("General options:") },
 { _("Limit framerate <= 60fps"), NULL, (int*)&display_cfg.limit_speed, 2,
{0, 1}, {_("No"),_("Yes")} },
 { _("Frame skip"), NULL, (int*)&display_cfg.frame_skip, 10, {0, 1, 2, 3,
4, 5, 6, 7, 8, 9},
   {_("Auto"), "1", "2", "3", "4", "5", "6", "7", "8", "9" } },
 { _("Screen rotation"), NULL, (int*)&display_cfg.user_rotate, 4, {0, 1, 2,
3 },
-  { _("None"), "90°", "180°", "270°" } },
+  { _("None"), "90Â°", "180Â°", "270Â°" } },
 { _("Flip screen"), NULL, (int*)&display_cfg.user_flip, 4, {0, 1, 2, 3 },
   { _("None"), _("Flip X"), _("Flip Y"), _("Flip XY") } },
 { _("Save per game screen settings"), NULL,
(int*)&raine_cfg.save_game_screen_settings, 2, {0, 1}, _("No"),_("Yes")},
 { _("Max FPS for drivers"), NULL, &ifps, ITEM_FLOATEDIT, { 10,0,180 }, {
"", (char*)&max_fps, "1", "200" } },
+{ _("Video information..."), &do_video, },
how can it guess that the last line here actually comes from the 1st
removed line ? It just can't, a human reader can because he understands the
code and sees similarities, but for the program it's too different here.

At least he marked the entries as fuzzy, so I reviewed a few fuzzy entries
from the french translation and there were some fixes to be made (+ a few
more extensions to do to this translation that I clearly stopped too early
!).

So to sum up : double check the fuzzy entries when an entry is moved and
not just modified, problem is that I don't have everyone, the only still
active translator is the brazilian portuguese one, he says that he has some
other fixes to do on his side so he will follow this on a few days.
For the rest I tried to fix a few (the video information mixed with the
video renderer entries), but I am sure to miss a few in this case.


Le mer. 28 déc. 2022 à 23:06, Emmanuel Anne ***@***.***> a
écrit :
